adobe/react-spectrum
# Implement scroll anchoring for overlay positioning

Closes https://github.com/orgs/adobe/projects/19/views/4?pane=issue&itemId=63064435

I think this fixes #3944 as well.

When an overlay repositions, its height might change, causing the focused item to go out of view. Even if the height didn't change, this might occur because we change the max-height before positioning in order to get the correct intrinsic size. This causes the browser to lose the previous scroll position. This happens during auto focus as well, when focusing happens prior to positioning.

This PR implements scroll anchoring. We save the offset of the focused element from the scroll container before positioning, and adjust the scroll position afterward so that it appears to remain in the same place relative to the viewport, even if the height of the scroll container changed.

**Test instructions**:

1. Open Picker story
2. resize the window to be small and select an item at the bottom
3. re-open the picker. Should scroll the auto focused item into view
4. Resize the height of the browser window. Focused item should remain anchored in view.
5. Repeat the same steps with RAC Select many items story.
